test/test_session.cpp

```diff
@@ -510,3 +510,70 @@ TORRENT_TEST(init_dht)
 		TEST_EQUAL(count, 1);
 	}
 }
+
+TORRENT_TEST(reopen_network_sockets)
+{
+	auto count_alerts = [](session& ses, int listen, int portmap)
+	{
+		int count_listen = 0;
+		int count_portmap = 0;
+		int num = 120; // this number is adjusted per version, an estimate
+		time_point const end_time = clock_type::now() + seconds(15);
```

Comment:
does it really take 15 seconds? I'm a bit worried about the amount of time this test takes to run.
If this was a simulation it wouldn't be a problem, but this is wall clock time.

Reply:
I will reduce both `num` and `end_time` to a minimum